src/tiledb/cloud/geospatial/helpers.py

```diff
@@ -0,0 +1,188 @@
+import math
+import os
+from dataclasses import dataclass
+from typing import Any, Iterator, Mapping, Optional, Sequence, Tuple, Union
+
+import pdal
+import rasterio
+import shapely
+from rtree import index
+
+import tiledb
+
+
+class RasterIngestionWarning(UserWarning):
+    pass
+
+@dataclass
+class GeoExtents:
+    minx: float
+    miny: float
+    maxx: float
+    maxy: float
+    minz: Optional[float] = None
+    maxz : Optional[float] = None
+    
+    @property
+    def bounds(self):
+        if self.minz:
+            return (self.minx, self.miny, self.minz,
+                    self.maxx, self.maxy, self.maxz)
+        else:
+            return (self.minx, self.miny, self.maxx, self.maxy)
+
+@dataclass
+class GeoMetadata:
+    path: os.PathLike
+    extents: GeoExtents
+    projection: str
```

Comment:
@normanb how about this?

```suggestion
    bbox: BoundingBox
    crs: str
```

`BoundingBox` is consistent with other uses of the term, including PDAL's and Rasterio's. `GeoExtents` could be renamed to `BoundingBox`, no need to change the class. `crs` is more strictly correct than projection.